### PR TITLE
Add Oracle Padding bug rule

### PR DESCRIPTION
**Pull request by ErebusZ**
Add Oracle Padding bug rule.

### CODE REVIEW

File: ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+        String data = "Jan van Eyck was here 1434";
+        String password = "ThisIs128bitSize";
+
+        final plainText = utf8.encode(data);
+        // specify an ecryption key and a random initialization vector
+        final key = Key.fromUtf8(password);
+        final iv = IV.fromLength(16);
+        // Encrypter instantiation with the CBC mode.
+        final encrypter = Encrypter(AES(key, mode: AESMode.cbc));
```
Comment (BlueSquare1) [inline ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart]:
```suggestion
        final encrypter = Encrypter(AES(key, mode: AESMode.cbc, padding: 'PKCS7'));
```
CBC algorithm itself is not inherently vulnerable to Oracle padding attacks, it is the padding scheme that is vulnerable.

**Review by BlueSquare1 — DISMISSED** [review]
CBC mode is not inherently vulnerable to Oracle padding, you need to add a padding algorithm

File: ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart (hunk 1 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart:23]: but the PKCS7 scheme is not the only padding scheme that is vulnerable to the oracle padding attack. both PKCS5 and ISO 10126 are also vulnerable. and also we currently raise this as info so it wouldnt hurt.

Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart]: good point!

File: ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'dart:convert';
+import 'package:encrypt/encrypt.dart';
+import 'package:ostorlab_insecure_flutter_app/bug_rule.dart';
+
+
+abstract class BugRule {
```
Comment (3asm) [inline ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart]: Rule not added to the main.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart (hunk 2 above)
Comment (ErebusZ) [reply ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart]: Added. [513ff50](https://github.com/Ostorlab/ostorlab_insecure_android_app/pull/19/commits/513ff509e5103fdd76e4514316513272325082c9)

**Review by 3asm — APPROVED** [review]

File: ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    String get description => 'This bug rule uses CBC insecure encryption mode.';
+
+    Future<void> run() async {
+
+        String data = "Jan van Eyck was here 1434";
+        String password = "ThisIs128bitSize";
+
+        final plainText = utf8.encode(data);
+        // specify an ecryption key and a random initialization vector
```
Comment (mohsinenar) [inline ostorlab_insecure_flutter_app/lib/bugs/oracle_padding.dart]:
```suggestion
        // Specify an ecryption key and a random initialization vector.
```

**Review by 3asm — APPROVED** [review]

**Pull request merged**